scribus/plugins/import/pdf/slaoutput.h

```diff
@@ -371,6 +462,63 @@ class SlaOutputDev : public OutputDev
 	QHash<QString, QList<int> > m_radioMap;
 	QHash<int, PageItem*> m_radioButtons;
 	int m_actPage;
+
+	//PDF Textbox framework
+	std::vector<TextRegion> m_textRegions = std::vector<TextRegion>();	
+};
+
+class AddFirstChar : public AddCharInterface

```

Comment:
the implementations of `Add*Char` are not in use yet, but i wonder if you could not just define them as function of the same class.

without seeing their usage it and the future plans it's hard to say, but fom the signature it's hard to say.

also:

- the implementation of the multiple `addChar()` seems to be missing
- if you stick to the multiple classes, i'm not sure that they should be defined in `slaouput.h`, but it's hard to say without seeing the big picture.

Reply:
there are 4 or 5 seperate wways of doing addchar and it get's messy if they arew all one function and that also slows things down and makes the code less readable.